messages/src/msg_types/protocols/pickup.rs

```diff
@@ -0,0 +1,32 @@
+use derive_more::{From, TryInto};
+use messages_macros::MessageType;
+use strum_macros::{AsRefStr, EnumString};
+use transitive::Transitive;
+
+use super::Protocol;
+use crate::msg_types::MsgKindType;
+
+#[derive(Copy, Clone, Debug, From, TryInto, PartialEq, MessageType)]
+#[msg_type(protocol = "messagepickup")]
+pub enum PickupType {
+    V2(PickupTypeV2),
+}
+
+#[derive(Copy, Clone, Debug, From, TryInto, PartialEq, Transitive, MessageType)]
+#[transitive(into(PickupType, Protocol))]
+#[msg_type(major = 2)]
+pub enum PickupTypeV2 {
+    #[msg_type(minor = 0, roles = "")]
```

Comment:
I know I was the one to say to set this to an empty string as there's no use in a dynamic discovery of protocols/roles an agent supports in terms of mediation, but adding a comment might be nice and it might clarify why no roles are added.

No strong opinion though, but I'd definitely do that.

Reply:
Perhaps it'd be suitable to have a mediator role there?

Mediator agents would perhaps advertise support for pickup protocol.

Reply:
Yeah, fair.

Reply:
Done